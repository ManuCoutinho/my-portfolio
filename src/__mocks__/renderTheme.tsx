import { ReactElement, ReactNode, FC } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
