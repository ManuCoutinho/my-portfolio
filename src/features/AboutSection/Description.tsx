import { StackBox, Subtitle } from 'components/Layout/Base'
import { Text } from 'components/Layout/Base'

export const Description = () => {
  return (
    <StackBox>
      <Subtitle>
        <span>Olá, sou</span> Manu Coutinho!
      </Subtitle>
      <Text>
        Eu sou uma desenvolvedora front-end, movida por uma paixão em construir
        coisas capazes de integrar diversos universos por meio da tecnologia.
      </Text>
      <Text>
        Gosto de desenvolver interfaces dinâmicas, de alta fidelidade e com
        designs pensados para transferir a melhor experiência ao usuário final.
      </Text>
      <Text>
        Minha abordagem para a resolução de problemas é altamente inclusiva,
        colaborativa e centrada no ser humano.
      </Text>
      <Text>Nesta página, irei compartilhar alguns dos meus projetos!</Text>
    </StackBox>
  )
}
