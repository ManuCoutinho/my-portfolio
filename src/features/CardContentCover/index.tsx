import { StackBox } from 'components/StackBox'
import dynamic from 'next/dynamic'

import { CardContentCoverProps } from './types'

const Slider = dynamic(() => import('components/Slider'))

const CardContentCover: React.FC<CardContentCoverProps> = ({ img }) => {
  return (
    <StackBox id='overview' direction='column'>
      <Slider img={img} />
    </StackBox>
  )
}

export default CardContentCover
