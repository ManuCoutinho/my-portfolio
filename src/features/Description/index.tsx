import { Heading } from 'components/Heading'
import { StackBox } from 'components/StackBox'
import { Text } from 'components/Text'

export const Description = () => {
  return (
    <StackBox direction='column'>
      <Heading as='h3' size='small'>
        <span>Olá, sou</span> Manu Coutinho!
      </Heading>
      <Text>
        Eu sou uma desenvolvedora front-end, movida por uma paixão em construir
        coisas capazes de integrar diversos universos por meio da tecnologia.
      </Text>
      <Text>
        Gosto de desenvolver interfaces dinâmicas, de alta fidelidade e com
        designs pensados para transferir a melhor experiência ao usuário final.
      </Text>
      <Text>
        Minha abordagem para a resolução de problemas é altamente inclusiva,
        colaborativa e centrada no ser humano.
      </Text>
      <Text>Nesta página, irei compartilhar alguns dos meus projetos!</Text>
    </StackBox>
  )
}
