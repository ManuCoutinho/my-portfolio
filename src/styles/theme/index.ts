import { DefaultTheme, CustomTheme } from 'styled-components'
import dark from './dark'
import light from './light'

export type Theme = typeof defaultTheme

const defaultTheme = {
	font: {
		cursive:
			"'Montserrat Alternates', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, 'Opens Sans', 'Helvetica Neue', sans-serif",
		regular:
			"'Josefin Sans', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, 'Opens Sans', 'Helvetica Neue', sans-serif"
	},
	fontSize: {
		xsm: '0.5rem', //8px
		sm: '0.875rem', //14px
		md: '1rem', //16px
		wd: '1.25rem', //20px
		lg: '1.5rem', //24px
		xl: '2rem', //36px
		xlg: '3rem' //48px
	},
	fontWeight: {
		thin: 100,
		light: 200,
		regular: 400,
		semibold: 600,
		bold: 700
	},
	radius: {
		sm: '8px',
		md: '16px',
		pill: '25px',
		full: '50%'
	}
} as const

function combineTheme(theme: CustomTheme): DefaultTheme {
	return { ...defaultTheme, ...theme }
}
export { combineTheme, dark, light }
