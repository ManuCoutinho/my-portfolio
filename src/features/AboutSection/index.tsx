import Image from 'next/image'
import { Description } from './Description'
import { Carousel } from './CarouselAbout'
import { StackBoxCenter, Title } from 'components/Layout/Base'
import { AboutSection, ContainerAbout } from './styles'
import { useTheme } from 'styled-components'

import lightImg from '../../../public/assets/perfil.webp'
import darkImg from '../../../public/assets/perfil-green.webp'

const About: React.FC = () => {
  const theme = useTheme()

  return (
    <AboutSection id='about'>
      <Title>Sobre</Title>
      <ContainerAbout>
        <StackBoxCenter>
          <Image
            width='300'
            height='300'
            src={theme.title === 'light' ? lightImg : darkImg}
            loading='lazy'
            alt='girl coding with cup of coffee'
          />
        </StackBoxCenter>
        <Description />
      </ContainerAbout>
      <Carousel />
    </AboutSection>
  )
}

export default About
