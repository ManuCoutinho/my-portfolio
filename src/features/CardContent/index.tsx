import dynamic from 'next/dynamic'
import * as Tabs from '@radix-ui/react-tabs'
import { Heading } from 'components/Heading'
import { StackBox } from 'components/StackBox'
import * as Styled from './styles'
import { CardContentProps } from './types'

const Overview = dynamic(() => import('features/CardContentCover'))
const Details = dynamic(() => import('features/CardContentDetails'))

export const CardContent: React.FC<CardContentProps> = ({ ...props }) => {
  return (
    <Tabs.Root defaultValue='overview'>
      <Styled.Header>
        <Heading as='h4' size='small'>
          {props.name}
        </Heading>
        <Tabs.List>
          <StackBox justify='space-evenly' gap={2}>
            <Styled.TabTitle value='overview'>Vis√£o geral</Styled.TabTitle>
            <Styled.TabTitle value='details'>Detalhes</Styled.TabTitle>
          </StackBox>
        </Tabs.List>
      </Styled.Header>
      <Styled.Content value='overview'>
        <Overview img={props.img} />
      </Styled.Content>
      <Styled.Content value='details'>
        <Details {...props} />
      </Styled.Content>
    </Tabs.Root>
  )
}
