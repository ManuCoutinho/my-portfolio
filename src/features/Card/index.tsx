import { useEffect, useState, FC } from 'react'
import { useRouter } from 'next/router'
import dynamic from 'next/dynamic'
import * as Dialog from '@radix-ui/react-dialog'
import { useTheme } from 'styled-components'
import { StackBox } from 'components/StackBox'
import { Heading } from 'components/Heading'
import { DataProps } from './types'
import * as Styled from './styles'
import { CardContent } from 'features/CardContent'

const Modal = dynamic(() => import('components/Dialog'), { ssr: false })

export const Card: FC<DataProps> = ({ ...props }) => {
  const router = useRouter()
  const { title } = useTheme()
  const [modalIsOpen, setModalIsOpen] = useState(false)
  const [isTouchDevice, setIsTouchDevice] = useState(false)

  const handleCloseModal = () => {
    router.push('/#portfolio')
  }

  useEffect(() => {
    if (typeof window !== 'undefined' && window.innerWidth <= 830) {
      setIsTouchDevice(true)
    }
  }, [router])
  function handleOpenDetails() {
    isTouchDevice
      ? router.push({
          pathname: `${props.id}`,
          query: { id: props.id }
        })
      : setModalIsOpen(true)
  }
  return (
    <StackBox>
      <Dialog.Root open={modalIsOpen} onOpenChange={setModalIsOpen}>
        <Styled.CardComponent type='button' onClick={handleOpenDetails}>
          <Styled.Image>
            <picture>
              <img
                loading='lazy'
                width={300}
                height={350}
                alt={props.alt}
                src={
                  title === 'light' ? props.coverAltLight : props.coverAltDark
                }
              />
              <source
                srcSet={
                  title === 'light'
                    ? props.coverDefaultLight
                    : props.coverDefaultDark
                }
                type='image/webp'
              />
            </picture>
          </Styled.Image>
          <Heading size='small' as='h3'>
            {props.name}
          </Heading>
          <Dialog.Trigger style={{ border: 0 }}>
            <Modal onClose={handleCloseModal}>
              <CardContent {...props} />
            </Modal>
          </Dialog.Trigger>
        </Styled.CardComponent>
      </Dialog.Root>
    </StackBox>
  )
}
