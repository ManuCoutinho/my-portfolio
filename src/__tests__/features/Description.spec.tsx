import { screen } from '@testing-library/react'
import { render } from '../../styles/theme/renderTheme'
import { Description } from '../../features/Description'

describe('<Description/>', () => {
  it('should render correctly', () => {
    render(<Description animate={true} />)
    const content = screen.getByRole('definition')
    const heading = screen.getByRole('heading')
    expect(content).toBeInTheDocument()
    expect(content).toContainElement(heading)
  })
  it('should match to snapshot', () => {
    const { container } = render(<Description animate={true} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
