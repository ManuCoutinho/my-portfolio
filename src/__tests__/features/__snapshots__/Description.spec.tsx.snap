// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Description/> should match to snapshot 1`] = `
<div
  class="sc-9a5cd56e-0 sc-12a7bdd2-0 ckLZLp iuSUVw"
  direction="column"
  role="definition"
>
  <h3
    class="sc-395b592-0 dENwrR"
  >
    <span>
      Olá, sou
    </span>
     Manu Coutinho!
  </h3>
  <p
    class="sc-6c8a52e-0 ffGzdG"
  >
    Eu sou uma desenvolvedora front-end, movida por uma paixão em construir coisas capazes de integrar diversos universos por meio da tecnologia.
  </p>
  <p
    class="sc-6c8a52e-0 ffGzdG"
  >
    Gosto de desenvolver interfaces dinâmicas, de alta fidelidade e com designs pensados para transferir a melhor experiência ao usuário final.
  </p>
  <p
    class="sc-6c8a52e-0 ffGzdG"
  >
    Minha abordagem para a resolução de problemas é altamente inclusiva, colaborativa e centrada no ser humano.
  </p>
  <p
    class="sc-6c8a52e-0 ffGzdG"
  >
    Aqui, compartillho alguns dos meus projetos!
  </p>
</div>
`;
